gymnasium_robotics/envs/multiagent_mujoco/coupled_half_cheetah.py

```diff
@@ -0,0 +1,84 @@
+import os
+
+import gymnasium
+import numpy
+import numpy as np
+from gymnasium.envs.mujoco import mujoco_env
+from gymnasium.utils.ezpickle import EzPickle
+
+
+class CoupledHalfCheetah(mujoco_env.MujocoEnv, EzPickle):
+    def __init__(self, agent_conf, render_mode: str = None):
+        self.metadata = {
+            "render_modes": [
+                "human",
+                "rgb_array",
+                "depth_array",
+            ],
+            "render_fps": 20,
+        }
+        observation_space = gymnasium.spaces.Box(
+            low=-numpy.inf, high=numpy.inf, shape=(35,), dtype=numpy.float32
+        )
+        mujoco_env.MujocoEnv.__init__(
+            self,
+            os.path.join(
+                os.path.dirname(os.path.abspath(__file__)),
+                "assets",
+                "coupled_half_cheetah.xml",
+            ),
+            5,
+            observation_space=observation_space,
+            render_mode=render_mode,
+        )
+        EzPickle.__init__(self)
+
+    def step(self, action):
+        xposbefore1 = self.unwrapped.data.qpos[0]
```

Comment:
I don't think you should need to use `unwrapped`, this should be equal to `self.data`